src/server/pps/server/api_server.go

```diff
@@ -1909,44 +1890,27 @@ func (a *apiServer) CreatePipelineInTransaction(
 	if err := a.authorizePipelineOpInTransaction(txnCtx, operation, newPipelineInfo.Details.Input, newPipelineInfo.Pipeline.Name); err != nil {
 		return err
 	}
-	update := false
-	if request.Update {
-		// check if the pipeline already exists, meaning this is a real update
-		if err := a.pipelines.ReadWrite(txnCtx.SqlTx).Get(request.Pipeline.Name, &pps.PipelineInfo{}); err == nil {
-			update = true
-		} else if !col.IsErrNotFound(err) {
-			return err
-		}
-	}
+	update := request.Update && oldPipelineInfo != nil
```

Comment:
So I guess you can set the update flag without an existing pipeline info?

Reply:
This is meant to be equivalent to the old logic, I think the idea was setting "update" means "I don't care if it already exists" (vs erroring in that case), while this variable is whether we need to perform cleanup operations on the existing version.